sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/implementation/AzureAuthorizedClientRepository.java

```diff
@@ -52,7 +52,7 @@ public void saveAuthorizedClient(OAuth2AuthorizedClient client,
             return (T) result;
         }
 
-        if (repo.isAuthzClient(id)) {
+        if (repo.isAuthzClient(id) && !repo.getAuthorizationProperties().get(id).getOnDemand()) {
```

Comment:
Can we update method `isAuthzClient`, add `getOnDemand` logic in method `isAuthzClient`?